### ISSUE TITLE
Support tiered storage

### ISSUE THREAD

**Issue by yuzhichang** [Altinity/clickhouse-operator#306]
Is it possible for the operator to expand existing ck instances' volume, or add another data volume?

**Comment by alex-zaitsev** [comment]
Currently it is not possible to extend PV if provider does not support it. If you need to extend, you have to add a new replica with PVC for a large size and then drop the old one. It is not automated, but we plan do add an automation in the future

**Comment by kushagra391** [comment]
can we currently attach multiple volume mounts ? 
like in this blog, https://www.altinity.com/blog/2019/11/29/amplifying-clickhouse-capacity-with-multi-volume-storage-part-2 ? (around `JBOD: Single-tier volume with several disks` section)

**Comment by yuzhichang** [comment]
@kushagra391 Yes. Here's the how to mount an existing PV to each clickhouse instance:
- Modify the manifest to mount another volume to `/var/lib/clickhouse2`:
```
diff --git a/docs/chi-examples/09-rolling-update-emptydir-01-initial-position.yaml b/docs/chi-examples/09-rolling-update-emptydir-01-initial-position.yaml
index 7db85d51..ab6dec51 100644
--- a/docs/chi-examples/09-rolling-update-emptydir-01-initial-position.yaml
+++ b/docs/chi-examples/09-rolling-update-emptydir-01-initial-position.yaml
@@ -20,8 +20,19 @@ spec:
               volumeMounts:
                 - name: clickhouse-storage
                   mountPath: /var/lib/clickhouse
+                - name: clickhouse-data-test
+                  mountPath: /var/lib/clickhouse2
           volumes:
             - name: clickhouse-storage
               emptyDir:
                 medium: "" # accepted values:  empty str (means node's default medium) or "Memory"
                 sizeLimit: 1Gi
+    volumeClaimTemplates:
+      - name: clickhouse-data-test
+        spec:
+          storageClassName: ""
+          accessModes:
+            - ReadWriteOnce
+          resources:
+            requests:
+              storage: 1Gi

```

- Destroy the clickhouse cluster, and apply again.

**Comment by kushagra391** [comment]
thanks @yuzhichang, thats helpful to know as well. I'll try that out. 

I was looking to extend (and not destroy->recreate) storage to multiple EBS volumes. 
Like suggested in the blog, 
```xml
-- Now let's try to join two SSD disks into one JBOD volume.  
-- We add the following storage policy to our storage.xml file: 

<ebs_gp2_jbod> > <!-- name for new storage policy -->
  <volumes>
    <ebs_gp2_jbod_volume> <!-- name of volume -->
       <!--
          the order of listing disks inside
          volume defines round-robin sequence
          -->
          <disk>ebs_gp2_1</disk>
          <disk>ebs_gp2_2</disk>
    </ebs_gp2_jbod_volume>
  </volumes>
</ebs_gp2_jbod>
```

usecase for it is, 
- one shard currently attaches to 1 EBS mounted at `/var/lib/clickhouse` 
- and max EBS storage is 16TB. 
- if we end up needing > 16TB storage, we probably need to add an extra clickhouse shard

so, was looking for ways to mount multiple EBS volumes to the same shard

**Comment by alex-zaitsev** [comment]
@kushagra391, here is an example of multi-volume configuration: https://github.com/Altinity/clickhouse-operator/blob/master/tests/configs/test-020-multi-volume.yaml

**Comment by kushagra391** [comment]
thanks @alex-zaitsev, thats super helpful !

**Comment by jmalicevic** [comment]
Hi,

I want to follow up on this post. I followed the above instructions but I keep having a 'there is no RW access to disk2' error. I want to note that I am running in a very secure setting, where users cannot run as root. But I have checked and /var/lib/clickhouse and /var/lib/clickhouse2 have the same owner and permission. However, files created in /var/lib/clickhouse2  are not executable: -rw-r--r-- .  
Excerpts from the YAML file: 

Everything runs as user 101 (which will be clickhouse) with the following: 
securityContext:
            seLinuxOptions:
              level: "s0"
            runAsGroup: 101
            runAsNonRoot: true
            runAsUser: 101 


settings:
      storage_configuration/disks/disk2/path: /var/lib/clickhouse2/
      storage_configuration/policies/default/volumes/default/disk: default
      storage_configuration/policies/default/volumes/disk2/disk: disk2

containers:
- name: data-storage-vc-template1
   mountPath: /var/lib/clickhouse
- name: disk2
   mountPath: /var/lib/clickhouse2

Thanks,
Jasmina
